classes/Dispatcher.php

```diff
@@ -228,30 +228,55 @@ protected function __construct()
 
         $this->loadRoutes();
     }
-
-    public function useDefaultController()
+    
+    /**
+     * Sets and returns the default controller.
+     *
+     * @param int $frontControllerType The front controller type
+     * @param Employee|null $employee The current employee
+     *
+     * @return string
+     */
+    private function setDefaultController($frontControllerType, Employee $employee = null)
     {
-        $this->use_default_controller = true;
-        if ($this->default_controller === null) {
-            if (defined('_PS_ADMIN_DIR_')) {
-                if (isset(Context::getContext()->employee)
-                    && Validate::isLoadedObject(Context::getContext()->employee)
-                    && isset(Context::getContext()->employee->default_tab)) {
-                    $this->default_controller =
-                        Tab::getClassNameById((int)Context::getContext()->employee->default_tab);
-                }
-                if (empty($this->default_controller)) {
-                    $this->default_controller = 'AdminDashboard';
+        switch ($frontControllerType) {
+            case self::FC_ADMIN:
+                // If there is an employee with a default tab set
+                if (null !== $employee) {
+                    $tabClassName = $employee->getDefaultTabClassName();
+                    if (null !== $tabClassName) {
+                        return $this->default_controller = $tabClassName;
+                    }
                 }
-            } elseif (Tools::getValue('fc') == 'module') {
-                $this->default_controller = 'default';
-            } else {
-                $this->default_controller = 'index';
-            }
+                // Default
+                return $this->default_controller = 'AdminDashboard';
+            case self::FC_MODULE:
+                return $this->default_controller = 'default';
+                break;
+            default:
+                return $this->default_controller = 'index';
```

Comment:
Another review would be interesting, but what about splitting the value affectation and the `return [...]; at the end of the method?